### PR TITLE
Fixes #1382

### LINKED ISSUE #1382: yarn init : error
**Do you want to request a *feature* or report a *bug*?**

Can't seem to execute `yarn init`.
This is the error I get: 
```
yarn init v0.16.1
error Command failed.
Exit code: 1
Command: git
Arguments: config user.name
Directory: /Users/user/Desktop/project
Output:

info Visit https://yarnpkg.com/en/docs/cli/init for documentation about this command.
```
**Please mention your node.js, yarn and operating system version.**
`Node version: v6.8.1 | Yarn version: v0.16.1 | Operating System:  macOS Sierra

### PR DESCRIPTION
**Pull request by jpoo90**
**Summary**

+ Added solution to fix #1382 . When the user doesn't have user in the `gitconfig` `yarn init` was crashing. 

**Test plan**

1. Remove the `user` property from your `gitconfig`. 
2. Run `yarn init`.
3. You'll be prompted with questions to add the `Author's name` & `Author's email`. (Both are optional)
4. The init process should be completed successfully.

**Before**

yarninitfailing

**After**

success

Feedback to improve this implementation is more than welcome 😄

### CODE REVIEW

File: src/cli/commands/init.js [hunk 1]
```diff
@@ -135,3 +134,16 @@ export async function run(
...
+
+async function getAuthorData(
+  reporter: Reporter,
+  credential: string,
+): Promise<string> {
+  try {
+    // try to get author default based on git config
+    return await child.spawn('git', ['config', credential]);
+  } catch (e) {
+    return await reporter.question(credential.replace('user.', 'author\'s '));
```
Comment (wyze) [inline src/cli/commands/init.js]: I think they prefer to double quote when there is a single quote instead of escaping the single quote.

**Comment by samccone** [comment]
@jpoo90 any interest in adding a test for this?

File: src/cli/commands/init.js [hunk 2]
```diff
@@ -135,3 +134,16 @@ export async function run(
...
+
+async function getAuthorData(
+  reporter: Reporter,
+  credential: string,
+): Promise<string> {
+  try {
+    // try to get author default based on git config
+    return await child.spawn('git', ['config', credential]);
+  } catch (e) {
+    return await reporter.question(credential.replace('user.', "author's"));
```
Comment (kittens) [inline src/cli/commands/init.js]: We should just return an empty string here. We already ask for the `author` field later on in the questionnaire.

**Comment by jpoo90** [comment]
@samccone Sure. I'm trying to figure out how to mock the `child.spawn('git', ['config'])` to make it fail in the test. Any recommendations will be really helpful.

**Comment by samccone** [comment]
@jpoo90 we should be able to just mock the return value of child.spawn for several conditions and that should work just swell.

**Comment by samccone** [comment]
awesome work here @jpoo90 I have built on top of it over here #1466 👏

**Comment by jpoo90** [comment]
I'm glad it helped, thanks for your help @samccone . Looking forward to contributing more.

**Pull request closed without merging**